src/components/z-pagination/index.tsx

```diff
@@ -0,0 +1,486 @@
+   */
+  @Watch("currentPage")
+  onPageChanged() {
+    this.pageChanged.emit(this.currentPage);
+    this.scrollToPage();
+  }
+
+  /**
+   * On split changed.
+   * Empty the `_visiblePages` value. The split feature wins over the pages chunks.
+   */
+  @Watch("split")
+  onSplitChanged() {
+    if (this.split) {
+      this._visiblePages = null;
+      this.edges = false;
+    }
+  }
+
+  /**
+   * Hide stuff if mobile.
+   */
+  @Listen("resize", { target: "window", passive: true })
+  checkScrollVisible() {
+    this.setMobile();
+  }
+
+  /**
+   *
+   * * @inheritdoc
+   */
+  componentDidRender() {
+    if (this.split) {
+      this._visiblePages = null;
+      this.edges = false;
+    }
+
+    this.setMobile();
+  }
+
+  /**
+   * Set functionalities according to screen size.
+   */
+  setMobile() {
+    const isMobileMediaQuery = "screen and (max-width: 768px)";
+    this.isMobile = window.matchMedia(isMobileMediaQuery).matches;
+  }
+
+  /**
+   * Get a list of pages chunks, each of `visiblePages` length.
+   * @returns {number[][]}
+   */
+  private getPagesChunks() {
+    // array of numbers from 1 to `totalPages`
+    const pages = Array.from({ length: this.totalPages }, (_, i) => i + 1);
+
+    if (this.split || this.isMobile) {
+      return [pages];
+    }
+
+    let chunks = [];
+    const chunksCount = Math.ceil(pages.length / this._visiblePages);
+    for (let index = 0; index < chunksCount; index++) {
+      chunks.push(
+        pages.slice(
+          index * this._visiblePages,
+          (index + 1) * this._visiblePages
+        )
+      );
+    }
+
+    return chunks;
+  }
+
+  /**
+   * Scroll to the left the chunk of pages containing the current page.
+   */
+  scrollToPage() {
+    const pageBtn = this.host.shadowRoot.querySelector(
+      `[data-page="${this.currentPage}"]`
+    ) as HTMLElement;
+
+    if (!pageBtn) {
+      return;
+    }
+
+    pageBtn.scrollIntoView({
+      behavior: "smooth",
+      block: "nearest",
+      inline: "center",
+    });
+  }
+
+  /**
+   * Select a page.
+   * Do validations on the passed value before assigning it to `currentPage`.
+   * @param {number} page Page number to set
+   */
+  selectPage(page) {
+    this.currentPage = Math.min(Math.max(page, 1), this.totalPages);
+  }
+
+  /**
+   * Event handler for go to page button click.
+   */
+  onGoToPage() {
+    const target = this.host.shadowRoot.querySelector(".gotopage-input") as HTMLZInputElement;
+    const newPage = Number(target.value);
+    if (newPage > 0) {
+      this.selectPage(newPage);
+    }
+  }
+
+  /**
+   * Render page number button.
+   * @param {number} page Page number to render
+   * @returns {HTMLButtonElement}
+   */
+  renderPage(page) {
+    return (
+      <button
+        class="page-button"
+        type="button"
+        aria-current={this.currentPage === page ? "page" : "false"}
+        title={`Vai alla pagina ${page}`}
+        data-page={page}
+        data-current={this.currentPage == page}
```

Comment:
In giro in questo file ci sono un tot di uguaglianze col doppio uguale. Farlo diventare ===?

Reply:
io di solito uso il doppio uguale quando ci sono dei numeri in mezzo, per evitare delle rotture di conversione se il numero arriva come stringa. in effetti qui il rischio è ridotto, metto il triplo